### PR TITLE
refactor: replace and remove toFormattedDate

### PR DESCRIPTION
**Pull request by MaelImhof**
This PR removes `DateTimeUtils.toFormattedDate` and replaces it with `DateTimeUtils.humanReadableFormat`.

`DateTimeUtils.toFormattedDate` used a date format that did not adapt to the current Locale. It used the `dd/MM/yyyy` format, but this format can be mistaken for `MM/dd/yyyy` depending on the date, which can be confusing for English speakers who are used to the second option rather than the first.

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt [hunk 1]
```diff
@@ -84,8 +83,3 @@ fun Timestamp.humanReadablePlannedLabel(
     else -> context.getString(R.string.datetime_utils_planned_on_date, formattedDate)
   }
 }
-
-fun Timestamp.toFormattedString(locale: Locale = Locale.getDefault()): String {
```
Comment (davidholzwarth) [inline app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt:88]: Question: IK its not really part of this PR specifically, but does `toHumanReadableFormat` also work for locales like Chinese and stuff? Could you help me understand what is going on in `humanReadableFormat`, because for me it low-key looks the same as the `toFormattedString` function

**Review by davidholzwarth — COMMENTED** [review]
Just a question

File: app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt (hunk 1 above)
Comment (MaelImhof) [reply app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt:88]: It is in fact the same as `toFormattedString`, the format being the main difference. `humanReadableFormat` uses `d MMMM yyyy` as format, which will produce dates such as (in English)

- 1 January 2001
- 17 April 2013

It spells out the complete name of the month. Of course, the format itself is not adapted to the locale, but the month is clearly spelled out instead of a number.

`toFormattedString` uses `dd/mm/yyyy`, which can easily be confused with `mm/dd/yyyy`, so I just thought it might be a good idea to use `humanReadableFormat` as well.

However I'm not sure how well it would work with Chinese locale, never tested it

Comment (davidholzwarth) [reply app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt:88]: ok, thats fine by me. Thanks!

**Review by davidholzwarth — APPROVED** [review]
LGTM fr this time
